packages/serverpod_serialization/test/auth_encoding_test.dart

```diff
@@ -0,0 +1,205 @@
+    ('Basic ', 'valid basic scheme with an empty base64 encoded value'),
+    ('basic dGVzd', 'valid basic scheme name in lowercase'),
+    ('BASIC dGVzd', 'valid basic scheme name in uppercase'),
+    ('basiC dGVzd', 'valid basic scheme name in mixed case'),
+  ];
+
+  /// standard non-basic auth schemes' key formats
+  var standardNonBasicAuthKeys = [
+    (
+      'Bearer dGVzd ,;.<>_-/\\|"\'`~!@#\$%^&*()+=[]{}',
+      'valid non-basic scheme'
+    ),
+    (
+      'bearer dGVzd ,;.<>_-/\\|"\'`~!@#\$%^&*()+=[]{}',
+      'valid non-basic scheme'
+    ),
+    (
+      'BEARER dGVzd ,;.<>_-/\\|"\'`~!@#\$%^&*()+=[]{}',
+      'valid non-basic scheme'
+    ),
+    (
+      'Digest dGVzd ,;.<>_-/\\|"\'`~!@#\$%^&*()+=[]{}',
+      'valid non-basic scheme'
+    ),
+    ('HOBA dGVzd ,;.<>_-/\\|"\'`~!@#\$%^&*()+=[]{}', 'valid non-basic scheme'),
+  ];
+
+  /// basic auth schemes' keys with invalid content
+  var invalidBasicAuthKeys = [
+    ('Basic dGVzd dGVzd dGVzd', 'invalid basic scheme with multiple values'),
+    ('Basic dGVzd dGVzd', 'invalid basic scheme with multiple values'),
+    ('Basic dGVzd ', 'invalid basic scheme with trailing space'),
+    ('Basic dGVz=d', 'basic scheme with invalid position of = character'),
+    ('Basic dGVzd*', 'basic scheme with invalid character in value'),
+    ('Basic d#', 'basic scheme with invalid character in value'),
+    ('Basic #', 'basic scheme with invalid character in value'),
+    ('Basic \t', 'basic scheme with invalid character in value'),
+    ('Basic  ', 'invalid basic scheme with space in place of value'),
+    ('Basic', 'invalid basic scheme with missing value'),
+  ];
+
+  /// arbitrary auth key formats
+  var arbitraryAuthKeys = [
+    ('', 'empty string is valid'),
+    (' ', 'space is valid'),
+    ('\t', 'white space is valid'),
+    ('\n', 'line break is valid'),
+    ('\r', 'line break is valid'),
+    (':', 'colon is valid'),
+    (':::::::::::::::', 'colon is valid'),
+    (',;.:<>_-/\\|"\'`~!@#\$%^&*()+=[]{}\r\n', 'special characters are valid'),
+    (
+      '%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%',
+      'long values are valid',
+    ),
+    (json.encode({'key': 'value'}), 'simple json-encoded values are valid'),
+    (
+      json.encode({
+        'key': 'value',
+        'key2': {
+          'key3': 'value3',
+          'key4': {
+            'key5': 'value5',
+            'key6': 'value6',
+            'key7': 'value7',
+            'key8': 'value8',
+            'key9': 'value9',
+            'key10': 'value10',
+          },
+        },
+        'key10': [],
+        'key11': [1, 2],
+        'key12': null,
+      }),
+      'complex json-encoded values are valid',
+    ),
+  ];
+
+  group('Auth header validity checking', () {
```

Comment:
Suggestion(NIT): Add a `-` or `:` symbol at the end of the grouping to clarify the prefix from the test name in the printed description.